astropy/coordinates/distances.py

```diff
@@ -142,13 +145,28 @@ def __new__(cls, value=None, unit=None, z=None, cosmology=None,
                 copy = False
 
             elif parallax is not None:
-                value = parallax.to(u.pc, equivalencies=u.parallax()).value
+                value = parallax.to_value(u.pc, equivalencies=u.parallax())
                 unit = u.pc
 
                 # Continue on to take account of unit and other arguments
                 # but a copy is already made, so no longer necessary
                 copy = False
 
+                if np.any(parallax < 0):
+                    if allow_negative:
+                        warnings.warn(
+                            "Negative parallaxes are converted to NaN "
+                            "distances even when `allow_negative=True`, "
+                            "because negative parallaxes cannot be transformed "
+                            "into distances. See discussion in this paper: "
+                            "https://arxiv.org/abs/1507.02105", AstropyWarning)
```

Comment:
Nice! An actual reference!

Reply:
Also note that I don't think `AstropyWarning` is actually necessary here (because it auto-becomes that).  Doesn't hurt, but thought I would mention.